CI failure due to `cabal test` picking a failing build plan

As seen in [this](https://github.com/GaloisInc/argo/actions/runs/9169375184/job/25209693644?pr=211#step:11:1) failing CI job:

```
$ cabal test argo
Resolving dependencies...
Error: cabal: Cannot test the package argo-0.1.0.0 because none of the
components are available to build: the test suite 'test-argo' is not available
because the solver picked a plan that does not include the test suites,
perhaps because no such plan exists. To see the error message explaining the
problems with such plans, force the solver to include the test suites for all
packages, by adding the line 'tests: True' to the 'cabal.project.local' file.

Error: Process completed with exit code 1.
```

That error message is confusing, but the root cause is relatively straightforward. The CI runs `cabal test` without the use of `--enable-tests`, which means that if `cabal`'s constraint solver finds competing build plans for `argo` where one build plan does allow the test suite dependencies and the other doesn't, then `cabal` is liable to pick the build plan that doesn't support the test suite (and indeed, that is what happens here). Something like this ought to fix it:

```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index d860134..106120c 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,6 +55,9 @@ jobs:
     - name: Typecheck python code
       working-directory: ./python
       run: poetry run mypy --install-types --non-interactive argo_client
+    - name: Configure
+      run: |
+        cabal configure --enable-tests
     - name: Build
       run: |
         cabal build all
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,6 +55,9 @@ jobs:
     - name: Typecheck python code
       working-directory: ./python
       run: poetry run mypy --install-types --non-interactive argo_client
+    - name: Configure
+      run: |
+        cabal configure --enable-tests
     - name: Build
       run: |
         cabal build all

